### PR TITLE
vim-patch:8.0.1239,8.1.{112,125,165,166,167,169,181,242,270,272,747,850,853,1519}

### PR DESCRIPTION
**Pull request by janlazo**

### CODE REVIEW

File: src/nvim/eval.c [hunk 1]
```diff
@@ -8891,6 +8913,7 @@ static void filter_map(typval_T *argvars, typval_T *rettv, int map)
       }
       hash_unlock(ht);
     } else {
+      // argvars[0].v_type == VAR_LIST
```
Comment (janlazo) [inline src/nvim/eval.c]: Why is this comment instead of an assert or `} else if { /* code for list type */ } else { /* throw error */ }`?

Comment (justinmk) [reply src/nvim/eval.c]: Vim almost never uses `assert()` AFAIR.

**Review by justinmk — APPROVED** [review]

File: src/nvim/eval.c [hunk 2]
```diff
@@ -955,6 +955,88 @@ eval_to_bool(
...
+  const int ret = eval1(arg, rettv, evaluate);
+  if (ret == FAIL) {
+    // Report the invalid expression unless the expression evaluation has
+    // been cancelled due to an aborting error, an interrupt, or an
+    // exception, or we already gave a more specific error.
+    // Also check called_emsg for when using assert_fails().
+    if (!aborting()
+        && did_emsg == did_emsg_before
+        && called_emsg == called_emsg_before) {
+      emsgf(_(e_invexpr2), arg);
```
Comment (janlazo) [inline src/nvim/eval.c]: This was `semsg` in the patch but we don't have one and the implementation looks similar to `emsgf`. For 8.1.0743, defining `semsg` as a macro or inline function to `emsgf` should suffice.

Comment (justinmk) [reply src/nvim/eval.c]: Agreed. Should be inline, otherwise a macro would be uppercase.

**Pull request merged**